python/observation-publisher/publisher/observability_backend.py

```diff
@@ -80,16 +31,111 @@ def __init__(
         self._model_version = model_version
         self._inference_schema = inference_schema
 
+    def common_arize_schema_attributes(self) -> dict:
+        return dict(
+            feature_column_names=self._inference_schema.feature_columns,
+            prediction_id_column_name=self._inference_schema.prediction_id_column,
+            timestamp_column_name=PREDICTION_LOG_TIMESTAMP_COLUMN,
+            tag_column_names=self._inference_schema.tag_columns,
+        )
+
+    def _preprocess_binary_classification(
+        self, df: pd.DataFrame
+    ) -> Tuple[pd.DataFrame, ArizeSchema]:
+        output_schema = self._inference_schema.binary_classification
+        prediction_label_column = "_prediction_label"
+        df[prediction_label_column] = df[output_schema.prediction_score_column].apply(
+            output_schema.prediction_label
+        )
+        schema_attributes = self.common_arize_schema_attributes() | dict(
+            prediction_label_column_name=prediction_label_column,
+            prediction_score_column_name=output_schema.prediction_score_column,
+        )
+        return df, ArizeSchema(**schema_attributes)
+
+    def _preprocess_score_categorical(
+        self, df: pd.DataFrame
+    ) -> Tuple[pd.DataFrame, ArizeSchema]:
+        output_schema = self._inference_schema.score_categorical
+        prediction_score_column = "_prediction_score"
+        prediction_label_column = "_prediction_label"
+        prediction_id_column = "_prediction_id"
+        prediction_tag_column = "_prediction_tag"
+        df[prediction_label_column] = df[output_schema.prediction_score_columns].apply(
+            lambda x: output_schema.prediction_label(x), axis=1
+        )
+        id_vars = [
+            c for c in df.columns if c not in output_schema.prediction_score_columns
+        ]
+        melted_df = df.melt(
+            id_vars=id_vars,
+            value_vars=output_schema.prediction_score_columns,
+            var_name=prediction_tag_column,
+            value_name=prediction_score_column,
+        )
+        melted_df[prediction_tag_column] = melted_df[prediction_tag_column].apply(
+            output_schema.prediction_tag
+        )
+        melted_df[prediction_id_column] = melted_df.apply(
+            lambda row: f"{row['prediction_id']}_{row[prediction_tag_column]}",
```

Comment:
how can you create this prediction_id_column during ground truth publishing?

Reply:
the problem is most of our users doesn't have prediction scores stored in the BQ, they need to modify their data pipelines to support this. Also if they have all the required data `features` + `prediction_scores` + `ground_truth` they won't need streaming publishing , they can use offline publishing.

Reply:
I agree to drop support for multiclass for now, since we don't have usecase, also arize plan to revamp their multiclass implementation. Maybe we can support later